src/main/java/edu/kit/provideq/toolbox/meta/ProblemType.java

```diff
@@ -46,12 +54,20 @@ public Class<ResultT> getResultClass() {
     return resultClass;
   }
 
+  /**
+   * Returns the bound estimator for this problem type.
+   */
+  public Function<InputT, Bound> getEstimator() {
```

Comment:
Can you refactor this to return Optional<Bound> instead to explicitly model the possibility of null?

Reply:
Sorry, instead the return value of the Function `Bound` should be optional, not the Function itself.